File: java/datastructure/linkedlist/leetcode/RemoveDuplicatesFromSortedList.java
package datastructure.linkedlist.leetcode;

/**
 * @author roseduan
 * @time 2020/11/17 12:17 下午
 * @description 删除排序链表中的重复元素
 */
public class RemoveDuplicatesFromSortedList {

    public ListNode deleteDuplicates(ListNode head) {
        ListNode p = head;
        while (p != null && p.next != null) {
            if (p.val == p.next.val) {
                p.next = p.next.next;
            } else {
                p = p.next;
            }
        }
        return head;
    }

    private static class ListNode {
        int val;
        ListNode next;
        ListNode(int x) { val = x; }
    }
}

File: python/linkedlist/leetcode/remove_duplicates_from_sorted_list.py

# 删除排序链表中的重复元素

class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:

    def deleteDuplicates(self, head: ListNode) -> ListNode:
        p = head
        while p and p.next:
            if p.val == p.next.val:
                p.next = p.next.next
            else:
                p = p.next
        return head
        